src/main/java/frc/robot/subsystems/ball/BallImpl.java

```diff
@@ -85,21 +86,22 @@ public void retract() {
     public void task() {
         if(this.shootEnabled) {
             this.shooter.runShooter();
-            this.feeder.enableHopper();
+            this.feeder.enable(FeederComponent.HOPPER);
             if(this.shooter.readyForBalls()) {
-                this.feeder.enableConveyor();
+                this.feeder.enable(FeederComponent.CONVEYOR);
             } else {
-                this.feeder.stopConveyor();
+                this.feeder.stop(FeederComponent.CONVEYOR);
             }
         } else {
             if(this.feeding == FeederStates.OFF){
                 this.shooter.stopShooter();
-                this.feeder.stopAll();
+                this.feeder.stop(FeederComponent.CONVEYOR);
+                this.feeder.stop(FeederComponent.HOPPER);
             } else if(this.feeding == FeederStates.UP) {
-                this.feeder.enableConveyor();
-                this.feeder.enableHopper();
+                this.feeder.enable(FeederComponent.CONVEYOR);
+                this.feeder.enable(FeederComponent.HOPPER);
             } else {
```

Comment:
Whether or not this becomes an enum, this should be a specific case or an `else if` clause. Right now it's effectively the default case of a switch, which as I explained in an earlier comment, can become confusing to debug if the enum values change.